pkg/plugin/framework/server_mux.go

```diff
@@ -75,3 +82,34 @@ func (m *serverMux) getHandler(name string) (interface{}, error) {
 
 	return m.handlers[name], nil
 }
+
+// ValidatePluginName checks if the given name:
+// - the plugin name has two parts separated by '/'
+// - non of the above parts is empty
+// - the prefix is a valid DNS subdomain name,
+// - a plugin with the same name does not already exist (if list of existing names is passed in)
+func ValidatePluginName(name string, existingNames []string) error {
```

Comment:
it'd be nice to have all of the validation errors that could be returned from this function have consistent phrasing about what a valid name is (`<DNS subdomain>/<non-empty name>`), and which part of that was invalid.

Reply:
Maybe something like the following:

```
- plugin name must have exactly two parts separated by a `/`. Accepted format: <DNS subdomain>/<non-empty name>
- both parts of the plugin name must be non-empty. Accepted format: <DNS subdomain>/<non-empty name>
- first part of the plugin name must be a valid DNS subdomain. Accepted format: <DNS subdomain>/<non-empty name>
```